tpgtools/documentation.go

```diff
@@ -48,28 +49,28 @@ func mergeResource(res *Resource, resources map[Version][]*Resource, version *Ve
 	return beta
 }
 
-func mergeResources(ga, beta *Resource) *Resource {
+func mergeResources(ga, beta *types.Resource) *types.Resource {
 	beta.Properties = mergeProperties(ga.Properties, beta.Properties)
 
 	return beta
 }
 
 // Marks any sub properties as beta only
-func mergeProperties(ga, beta []Property) []Property {
-	gaProps := make(map[string]Property)
+func mergeProperties(ga, beta []types.Property) []types.Property {
+	gaProps := make(map[string]types.Property)
 	for _, p := range ga {
-		gaProps[p.title] = p
+		gaProps[p.Name()] = p
 	}
-	betaProps := make(map[string]Property)
+	betaProps := make(map[string]types.Property)
 	for _, p := range beta {
-		betaProps[p.title] = p
+		betaProps[p.Name()] = p
```

Comment:
this might need to be looked at

Reply:
Yeah, title and name are slightly different:

```
// Name is the shortname of a field. For example, "machine_type".
func (p Property) Name() string {
	if len(p.customName) > 0 {
		return p.customName
	}
	return p.title
}
```

But I'm unsure on which of these is the correct one to use here

Reply:
I think `Name` is correct here, we're using the Terraform field name to uniquely identify the fields, right?